### ISSUE TITLE
ML-Tuner: Train on real profiling data instead of heuristic weights

### ISSUE THREAD

**Issue by noahgift** [paiml/trueno#80]
## Problem

The `BrickTuner` currently uses hardcoded heuristic weights derived from SHOWCASE-BRICK-001 analysis rather than being trained on actual profiling data.

```rust
// Current: hardcoded weights
weights[7] = 0.3; // batch_size_norm
weights[9] = 0.1; // cuda_graphs
weights[36] = 0.15; // gpu_mem_bw_norm
```

## Proposed Solution

1. Implement `TunerDataCollector` to automatically record profiling runs
2. Store training data in SQLite/Parquet format
3. Implement batch training pipeline using `aprender` RandomForest
4. Bootstrap from Five-Whys archive (85 labeled iterations)

## Acceptance Criteria

- [ ] Training data collector records BrickProfiler runs automatically
- [ ] Minimum 1000 samples before model training triggers
- [ ] MAPE < 10% on holdout test set (F001 falsification)
- [ ] R² > 0.85 on throughput prediction (F002 falsification)

## References

- Spec: `docs/specifications/ml-tuner-bricks.md` Section 5
- TUNER-010, TUNER-011 tickets in spec

**Comment by noahgift** [comment]
Implemented. `TunerDataCollector::record()` captures real profiling data as `TrainingSample` structs (features, throughput, timestamp, kernel_type, bottleneck_class). Auto-retrain threshold: MIN_SAMPLES_FOR_TRAINING=1000. Connected to BrickTuner ensemble. 20+ tests.
